mediapicker/src/main/java/org/wordpress/android/mediapicker/viewmodel/MediaPickerViewModel.kt

```diff
@@ -600,38 +648,125 @@ internal class MediaPickerViewModel @Inject constructor(
 
     @Suppress("NestedBlockDepth")
     private fun buildSoftAskView(softAskRequest: SoftAskRequest?): SoftAskViewUiModel {
+        return if (softAskRequest?.types?.contains(IMAGES) == true ||
+            softAskRequest?.types?.contains(PermissionsRequested.VIDEO) == true ||
+            softAskRequest?.types?.contains(PermissionsRequested.AUDIO) == true) {
+            if (VERSION.SDK_INT >= VERSION_CODES.TIRAMISU) {
+                buildMediaSoftAskRequest(softAskRequest)
+            } else {
+                throw UnsupportedOperationException(
+                    "Unsupported permissions for SDK < 33: ${softAskRequest.types.joinToString()}"
+                )
+            }
+        } else {
+            buildOldSoftAskView(softAskRequest)
+        }
+    }
+
+    // Requests READ_EXTERNAL_STORAGE on devices running SDK < 33
+    @Suppress("NestedBlockDepth")
+    private fun buildOldSoftAskView(softAskRequest: SoftAskRequest?): SoftAskViewUiModel {
```

Comment:
"buildOldSoftAskView" and "buildMediaSoftAskRequest" share many common lines of code. What do you think about merging them? Have you considered it?"

Reply:
I took a closer look at this and I've done a bit more refactoring to simplify things. I just need to fix some edge cases and test things properly so please hang tight :).